src/main/scala/lc/core/config.scala

```diff
@@ -44,25 +44,9 @@ object Config {
     case JField("config", JObject(config)) => ("config", JString(config.toString))
   }
   val captchaConfig: List[CaptchaConfig] = captchaConfigTransform.extract[List[CaptchaConfig]]
-  val allowedLevels: Set[String] = getAllValues(configJson, ParametersEnum.ALLOWEDLEVELS.toString)
-  val allowedMedia: Set[String] = getAllValues(configJson, ParametersEnum.ALLOWEDMEDIA.toString)
-  val allowedInputType: Set[String] = getAllValues(configJson, ParametersEnum.ALLOWEDINPUTTYPE.toString)
-
-  private def getAllValues(config: JValue, param: String): Set[String] = {
-    val configValues = (config \\ param)
-    val result = for {
-      JObject(child) <- configValues
-      JField(param) <- child
-    } yield (param)
-
-    var valueSet = Set[String]()
-    for (valueList <- result) {
-      for (value <- valueList._2.children) {
-        valueSet += value.values.toString
-      }
-    }
-    valueSet
-  }
+  val allowedLevels: Set[String] = captchaConfig.flatMap(_.allowedLevels).toSet
+  val allowedMedia: Set[String] = captchaConfig.flatMap(_.allowedMedia).toSet
+  val allowedInputType: Set[String] = captchaConfig.flatMap(_.allowedInputType).toSet
```

Comment:
@rr83019 Is this ok? Instead of reparsing the JSON, I am reusing the data already available in `captchaConfig`.

The JSON parsing was failing when only one Captcha was specified in the config.

Reply:
Yeah, looks good!